pkg/portfolio/service_test.go

```diff
@@ -35,8 +35,8 @@ func TestGetPortfolioByID(t *testing.T) {
 
 	// Portfolio doesn't exist: Test to check when error is returned from repo
 	// error is returned from service
-	mockRepo.EXPECT().GetPortfolioByID("1").Return(portfolio.Portfolio{}, portfolio.ErrMissing).Times(1)
-	p, err := s.GetPortfolioByID("1")
+	mockRepo.EXPECT().Portfolio("1").Return(portfolio.Portfolio{}, portfolio.ErrMissing).Times(1)
+	p, err := s.Portfolio("1")
 	if err != portfolio.ErrMissing {
 		t.Errorf("Portfolio doesn't exist: Error missing not returned")
```

Comment:
I'd recommend using some sort of matcher library to clear up these 3 lines per error. My recommendation is https://github.com/stretchr/testify .

Basically, you can have `assert.Equal(t, err, portfolio.ErrMissing)`. It reads just as it implies. The opposite is `asser.NoError(t, err)` and can be used for the tests too.

In test setups, you can use `require.NoError(t, err)` which will stop a test if it fails. Assert allows the test to continue to run and then checks if all asserts were met.